src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpVirtualDocument.cs

```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
 using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.VisualStudio.LanguageServerClient.Razor
```

Comment:
In this class note the `TextBuffer.SetHostDocumentSyncVersion(_hostDocumentSyncVersion.Value);` and the `MakeEmptyEdit` hack. I don't love how we've marked our buffers with their host document versions to detect synchronization (as mentioned above in this PR) super open for other ideas on how to generalize this

Reply:
Sounds good, will think about it.

---
In reply to: [437131550](https://github.com/dotnet/aspnetcore-tooling/pull/1994#discussion_r437131550) [](ancestors = 437131550)